docs/docs/gatsby-image-api.md

```diff
@@ -0,0 +1,344 @@
+---
+title: Gatsby Image API
```

Comment:
I like the title of "Gatsby Image API", but I think the URL should be gatsbyjs.org/docs/gatsby-image — maybe rename this to `gatsby-image.md`?

Reply:
Sure, we can do that. I picked that URL to differentiate from other files that already exist, but it isn't consistent with the others.